controllers/llamastackdistribution_controller_test.go

```diff
@@ -312,3 +320,132 @@ func TestReconcile(t *testing.T) {
 	AssertResourceOwnedByInstance(t, networkpolicy, instance)
 	AssertResourceOwnedByInstance(t, serviceAccount, instance)
 }
+
+// Define a custom roundtripper type for testing.
+type mockRoundTripper struct {
+	RoundTripFunc func(req *http.Request) (*http.Response, error)
+}
+
+// RoundTrip satisfies the http.RoundTripper interface and calls the mock function.
+func (m *mockRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	return m.RoundTripFunc(req)
+}
+
+// newMockAPIResponse is a test helper that takes any data structure,
+// marshals it to JSON, and returns a complete http response.
+func newMockAPIResponse(t *testing.T, data any) *http.Response {
+	t.Helper()
+	jsonBytes, err := json.Marshal(data)
+	require.NoError(t, err)
+
+	return &http.Response{
+		StatusCode: http.StatusOK,
+		Body:       io.NopCloser(strings.NewReader(string(jsonBytes))),
+		Header:     http.Header{"Content-Type": []string{"application/json"}},
+	}
+}
+
+func TestLlamaStackProviderAndVersionInfo(t *testing.T) {
+	// arrange
+	enableNetworkPolicy := false
+	expectedLlamaStackVersionInfo := "v-test"
+	expectedProviderID := "mock-ollama"
+
+	// define the data structure for the mock providers response
+	providerData := struct {
+		Data []llamav1alpha1.ProviderInfo `json:"data"`
+	}{
+		Data: []llamav1alpha1.ProviderInfo{
+			{
+				ProviderID:   expectedProviderID,
+				ProviderType: "remote::ollama",
+				API:          "inference",
+				Health:       llamav1alpha1.ProviderHealthStatus{Status: "OK", Message: ""},
+				Config:       apiextensionsv1.JSON{Raw: []byte(`{"url": "http://mock.server"}`)},
+			},
+		},
+	}
+
+	// define the data structure for the mock version response
+	versionData := struct {
+		Version string `json:"version"`
+	}{
+		Version: expectedLlamaStackVersionInfo,
+	}
+
+	// create the mock http client that uses our custom roundtripper
+	mockClient := &http.Client{
+		Transport: &mockRoundTripper{
```

Comment:
Could add a timeout here to match the non-test client (if you want to):

```suggestion
	mockClient := &http.Client{
	        Timeout: 5 * time.Second,
		Transport: &mockRoundTripper{
```